kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java

```diff
@@ -0,0 +1,67 @@
+package org.kie.workbench.drools.client.docks;
+
+import com.google.gwt.user.client.Window;
+import org.kie.workbench.common.screens.datamodeller.client.DataModelerContext;
+import org.kie.workbench.common.screens.datamodeller.client.context.DataModelerWorkbenchContext;
+import org.kie.workbench.common.screens.datamodeller.client.context.DataModelerWorkbenchContextChangeEvent;
+import org.uberfire.client.workbench.docks.UberfireDock;
+import org.uberfire.client.workbench.docks.UberfireDockPosition;
+import org.uberfire.client.workbench.docks.UberfireDockReadyEvent;
+import org.uberfire.client.workbench.docks.UberfireDocks;
+import org.uberfire.mvp.PlaceRequest;
+import org.uberfire.mvp.impl.DefaultPlaceRequest;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+@Dependent
+public class AuthoringWorkbenchDocks {
+
+    @Inject
+    private UberfireDocks uberfireDocks;
+
+    @Inject
+    protected DataModelerWorkbenchContext dataModelerWBContext;
+
+    private String authoringPerspectiveIdentifier;
+
+    private UberfireDock projectExplorerDock;
+
+
+    public void perspectiveChangeEvent(@Observes UberfireDockReadyEvent dockReadyEvent) {
+        if (authoringPerspectiveIdentifier != null && dockReadyEvent.getCurrentPerspective().equals(authoringPerspectiveIdentifier)) {
+            if (projectExplorerDock != null) {
+                uberfireDocks.expand(projectExplorerDock);
```

Comment:
This was necessary to open(expand) the project explorer dock every time that we open a perspective.

The authoring doesn't work without that -> (http://prntscr.com/862cnq)